### PR TITLE
Transformation Controls - Rotation

### PR DESCRIPTION
**Pull request by arjxn-py**

### CODE REVIEW

File: packages/base/src/3dview/mainview.tsx [hunk 1]
```diff
@@ -442,58 +442,61 @@ export class MainView extends React.Component<IProps, IStates> {
...
-              pivotEuler.y * matchingChildEuler.y,
-              pivotEuler.z * matchingChildEuler.z,
-              pivotEuler.order // Ensure we use the same rotation order
+        } else if (this._transformControls.mode === 'rotate' && this._pivot) {
+            // Retrieve the existing rotation from the shared model
+            const existingRotationArray = obj?.parameters?.Placement?.Axis;
+            const existingRotation = new THREE.Euler(
+                THREE.MathUtils.degToRad(existingRotationArray[0]),
+                THREE.MathUtils.degToRad(existingRotationArray[1]),
+                THREE.MathUtils.degToRad(existingRotationArray[2]),
```
Comment (trungleduc) [inline packages/base/src/3dview/mainview.tsx]: The Placement axis is not the Euler rotation angle. You could check the definition here https://wiki.freecad.org/Placement

File: packages/base/src/3dview/mainview.tsx [hunk 2]
```diff
@@ -422,20 +422,68 @@ export class MainView extends React.Component<IProps, IStates> {
...
+            const finalQuaternion =
+              existingQuaternion.multiply(pivotQuaternion);
+
+            const angle = 2 * Math.acos(finalQuaternion.w);
+            const angleDeg = THREE.MathUtils.radToDeg(angle);
+            const axis = new THREE.Vector3();
+            axis.set(finalQuaternion.x, finalQuaternion.y, finalQuaternion.z);
+
+            // Update the shared model with the new axis and angle
+            this._model.sharedModel.updateObjectByName(objectName, {
```
Comment (martinRenou) [inline packages/base/src/3dview/mainview.tsx:470]: This should be using the new `this._mainViewModel.maybeUpdateObjectParameters`

File: packages/base/src/3dview/mainview.tsx [hunk 3]
```diff
@@ -1173,12 +1224,12 @@ export class MainView extends React.Component<IProps, IStates> {
...
       const selectedMeshName = selection[0];
-      const matchingChild = this._meshGroup?.children.find(child =>
+      this._matchingChild = this._meshGroup?.children.find(child =>
         child.name.startsWith(selectedMeshName)
       );
 
-      if (matchingChild) {
-        this._transformControls.attach(matchingChild as BasicMesh);
+      if (this._matchingChild) {
+        this._transformControls.attach(this._matchingChild as BasicMesh);
```
Comment (martinRenou) [inline packages/base/src/3dview/mainview.tsx:1232]: I'm surprised you're still attaching the matchingchild to the transform control. If you can can make it work with attaching the matchingchild, isn't the pivot object useless?

File: packages/base/src/3dview/mainview.tsx (hunk 2 above)
Comment (arjxn-py) [reply packages/base/src/3dview/mainview.tsx:470]: for rotation the controls have been attaching to pivot only [see here](https://github.com/jupytercad/JupyterCAD/pull/574/files/e78d8c8954513076a5c529ee0ed9e7c6f2807dbb#diff-37a7f3f4d02e2bc4c4f40a612b75d24569d7c0c46420298e3281f4af6e29b7e6R1252)

For translation i've kept the logic straight and same as before

File: packages/base/src/3dview/mainview.tsx (hunk 3 above)
Comment (arjxn-py) [reply packages/base/src/3dview/mainview.tsx]: for rotation the controls have been attaching to pivot only [see here](https://github.com/jupytercad/JupyterCAD/pull/574/files/e78d8c8954513076a5c529ee0ed9e7c6f2807dbb#diff-37a7f3f4d02e2bc4c4f40a612b75d24569d7c0c46420298e3281f4af6e29b7e6R1252)

For translation i've kept the logic straight and same as before

File: packages/base/src/3dview/mainview.tsx [hunk 4]
```diff
@@ -1167,40 +1200,53 @@ export class MainView extends React.Component<IProps, IStates> {
...
             positionArray[0],
             positionArray[1],
             positionArray[2]
           );
-          this._transformControls.position.copy(positionVector);
+          const rotation = new THREE.Vector3(
+            axis[0],
+            axis[1],
+            axis[2]
+          ).multiplyScalar(THREE.MathUtils.degToRad(angle));
```
Comment (martinRenou) [inline packages/base/src/3dview/mainview.tsx:1234]:
```suggestion
          ).normalize().multiplyScalar(THREE.MathUtils.degToRad(angle));
```

**Comment by arjxn-py** [comment]
Replaced by #603

**Pull request closed without merging**